src/Build/BackEnd/Components/Logging/ProjectLoggingContext.cs

```diff
@@ -71,6 +72,17 @@ internal ProjectLoggingContext(NodeLoggingContext nodeLoggingContext, BuildReque
         {
         }
 
+        private IEnumerable<DictionaryEntry> FilterEnvironmentDerivedProperties(PropertyDictionary<ProjectPropertyInstance> properties)
+        {
+            foreach (ProjectPropertyInstance property in properties)
+            {
+                if (property is not EnvironmentDerivedProjectPropertyInstance)
+                {
+                    yield return new DictionaryEntry(property.Name, property.EvaluatedValue);
```

Comment:
I'm also thinking whether an iterator is risky here. Always feels safer to allocate and return (what if someone iterates over this several times? what if the underlying properties collection changes before or while we iterate?) I don't know where this will be realized. 

On the other hand allocating eagerly will allocate, while in theory an iterator will have a fixed allocation cost. Not sure really...

Reply:
This seems to be called as part of logging that the build started in RequestBuilder.BuildProject. I would imagine that the _requestEntry is stable before that's called? If so, that would make this safe; I'm not at all sure, though.